src/task/callback.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+//! A [`FnOnce`] or [`FnMut`] closure.
+
+use crate::LocalSpawn;
+
+use std::marker::PhantomData;
+
+/// Tasks that is supported by [`Runner`].
+///
+/// Anything that can be converted to a [`Task`] is supported by [`Runner`].
+pub trait CallbackTask<Spawn> {
```

Comment:
Is it still necessary given that the queue definition is hidden behind `*Spawn` trait?

Reply:
The problem is that `LocalSpawn` sometimes needs extra information besides the original task.

For example, if we want to spawn a new task to the `LocalSpawn` of a multi-level queue, at least a counter recording the task's running time needs to be passed in along with the task.

Reply:
I remove this trait and add a `TaskContext` associated type to `LocalSpawn`. I think now it's flexible enough.

Resemble to `TaskContext`, I also add `SpawnOption` to `RemoteSpawn`. Currently in my mind, it can be used to specify the priority (nice) of a task.